File: Hackerrank/Drawing_Book.rb
def pageCount(n, p)
    if ( n.to_f / 2 >= p )
        answer = p.to_f / 2
    else
        new_num = n - p
        answer = (new_num.to_f / 2)
        answer += 1 if ( n % 2 == 0 && p % 2 != 0 )
    end
    answer.floor
end

File: Hackerrank/Drawing_Book.js
function pageCount(n, p) {
    let answer;
    if (n / 2 > p) {
        answer = Math.floor(p / 2)
    } else {
        let newNum = n - p;
        if (p % 2 !== 0) {
            answer = Math.floor(newNum / 2) + 1;
        } else {
            answer = Math.floor(newNum / 2);
        }
    }
    return answer;
}
